src/app/components/datacard/datacard.component.ts

```diff
@@ -26,16 +26,16 @@ export class DatacardComponent implements OnInit, OnDestroy {
     constructor(private covid: CovidService, private router: ActivatedRoute, private adapter: CountryReportAdapter) {}
 
     ngOnInit(): void {
-        this.stateHandler("init")
+        this.stateInit()
         this.suscription = this.covid.getCountriesSummary().subscribe((countries: CountryApi[]) => {
```

Comment:
This subscription should be a function. Let's keep OnInit cleaner

Reply:
migration of stateHandler to a service done. also some refactoring in datacard.component.ts as requested.